lib/cocoapods/downloader/cache.rb

```diff
@@ -304,55 +279,23 @@ def in_tmpdir(&blk)
       #
       # @return [Void]
       #
-      def copy_source_and_clean(source, destination, spec)
-        specs_by_platform = group_subspecs_by_platform([spec])
+      def copy_and_clean(source, destination, spec)
+        specs_by_platform = group_subspecs_by_platform(spec)
         destination.parent.mkpath
         Cache.write_lock(destination) do
-          Pod::Executable.execute_command('rsync', ['-a', '--exclude=.git/fsmonitor--daemon.ipc', '--delete', "#{source}/", destination])
+          FileUtils.rm_rf(destination)
+          FileUtils.cp_r(source, destination)
           Pod::Installer::PodSourcePreparer.new(spec, destination).prepare!
           Sandbox::PodDirCleaner.new(destination, specs_by_platform).clean!
         end
       end
 
-      # Copies the `files` from the `source` directory to `destination` _without_ cleaning the
-      # `destination` directory of any files unused by `spec`. This is a faster version used when
-      # installing pods without a prepare command which has already happened prior.
-      #
-      # @param  [Array<Pathname>] files
-      #
-      # @param  [Pathname] source
-      #
-      # @param  [Pathname] destination
-      #
-      # @return [Void]
-      #
-      def copy_files(files, source, destination)
-        files = files.select { |f| File.exist?(f) }
-        destination.parent.mkpath
-        Cache.write_lock(destination) do
-          FileUtils.rm_rf(destination)
-          destination.mkpath
-          files_by_dir = files.group_by do |file|
-            relative_path = Pathname(file).relative_path_from(Pathname(source)).to_s
-            destination_path = File.join(destination, relative_path)
-            File.dirname(destination_path)
-          end
-
-          files_by_dir.each do |dir, files_to_copy|
-            FileUtils.mkdir_p(dir)
-            FileUtils.cp_r(files_to_copy, dir)
```

Comment:
Looks like the issue is that when this line copies a link and the link points to already existing file then it trow the error. cc @dnkoutso